app/controllers/cases/offender_sar_controller.rb

```diff
@@ -212,15 +212,37 @@ def confirm_sent_to_sscl
       redirect_to case_path(@case) and return
     end
 
-    def confirm_information_received
-      information_received = params[:offender_sar].try(:[], :information_received)
-      return if information_received.present?
+    def accepted_date_received
+      @case = Case::Base.find_by(id: params[:id])
+      @case.received_date_dd = ""
+      @case.received_date_mm = ""
+      @case.received_date_yyyy = ""
+    end
 
-      @case.errors.add(
-        :information_received,
-        I18n.t("activerecord.errors.models.case.attributes.information_received.blank"),
-      )
-      render :information_received and return
+    def confirm_accepted_date_received
+      @case = Case::Base.find_by(id: params[:id])
+      authorize @case, :can_edit_case?
+
+      received_date = params[:offender_sar].permit(:received_date_dd, :received_date_mm, :received_date_yyyy)
+      service = case_updater_service.new(current_user, @case, received_date)
+      service.call
+      @case.number.tr!("R", "")
```

Comment:
This shouldn't be done here. It would be better to create a service to:
- Update the case
- Change the case number
- Transition the state
- Add any history messages